### PR TITLE
Servo as local planner demonstration

### PR DESCRIPTION
**Pull request by AdamPettinger**
# Description

- Adds Servo as a local planner (following @sjahr's PR for that over in moveit2)
- Hardcodes the MTC global planner to do a relative Cartesian plan
- Adds a joint configuration for the start of the Cartesian path
- Allows Servo local planner to execute the trajectory at constant velocity, with user input to allow rotation of the last joint on-line

# How to test
Follow the updated README section `Hybrid Planning Demonstration`

Note this PR includes a dependency change in `moveit2`, so you should pull the correct version and build in order to test properly

# Next Steps
This is mostly a demonstration of the hybrid planning capabilities. I am not too sure we actually want to merge this in (or at least as is now with hardcoded stuff for the demo). @henningkayser do you have any next steps for this effort?

### CODE REVIEW

**Review by henningkayser — COMMENTED** [review]
I think this is a really useful step and we should merge this so that @christianlandgraf can reproduce this with hardware. I'd be interested to see if we see any performance differences. @AdamPettinger how about adding your temporary local solver to the demo package next to the global MTC solver? That way we can continue experimenting with it, possibly adding IK-based jogging as a next step?

**Comment by AdamPettinger** [comment]
I am working on adding it, don't quite have the pluginlib configurations correct yet so the local planner does not find the new `ServoSolver` plugin

**Comment by christianlandgraf** [comment]
Looks great!! Unfortunately, the `hybrid_planning_container`  fails on my setup although the global and local goals seem to be succesfully accepted... Somehow it worked yesterday with commit 0fa9dbebc702fe659d64994a554c2e96a90b7559 but I didn't manage to get it working again and therefore I didn't tested it on real hardware.

Besides that, I had to do some minor fixes to succesfully compile everything:
- Switch to `foxy` branch in `geometric_shapes` (in [upstream.repos](https://github.com/PickNikRobotics/UR10e_welding_demo/blob/hackathon/upstream.repos#L18) ). I think this is one reason for the current build fail of CI
- Merge the current `main` branch of moveit2 into the `feature/hybrid_planning` branch in https://github.com/henningkayser/moveit2/tree/feature/hybrid_planning. 
- Add `/` symbols to the end of each include directory for header files  in [this CMakeLists.txt](https://github.com/henningkayser/moveit2/blob/feature/hybrid_planning/moveit_ros/hybrid_planning/CMakeLists.txt#L50)

**Comment by AdamPettinger** [comment]
The latest commit I just added is following the [ROS 2 Pluginlib](https://ftp.osuosl.org/pub/ros/ros_docs_mirror/en/foxy/Tutorials/Pluginlib.html) tutorial with the local solver in our (`hybrid_planning_demo`) package. However, this still isn't loading the plugins correctly, see error below. If I could ask maybe @sjahr to take a look and see if there are any glaring problems, that would be great

> [ERROR - local_planner_component]: Exception while loading trajectory operator 'hybrid_planning_demo::SimpleServoSampler': 'Could not find library corresponding to plugin hybrid_planning_demo::SimpleServoSampler. Make sure the plugin description XML file has the correct name of the library and that the library actually exists.'
[ERROR - local_planner_component]: Exception while loading constraint solver 'hybrid_planning_demo::ServoSolver': 'Could not find library corresponding to plugin hybrid_planning_demo::ServoSolver. Make sure the plugin description XML file has the correct name of the library and that the library actually exists.'

File: hybrid_planning_demo/plugins.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<library path="hybrid_planning_demo">
+  <class type="hybrid_planning_demo::ServoSolver" base_class_type="moveit_hybrid_planning::LocalConstraintSolverInterface">
+    <description>Uses Servo as a local solver.</description>
+  </class>
+  <class type="hybrid_planning_demo::SimpleServoSampler" base_class_type="moveit_hybrid_planning::TrajectoryOperatorInterface">
+    <description>Remake of SimpleSampler to use with Servo local solver.</description>
+  </class>
+</library>
```
Comment (sjahr) [inline hybrid_planning_demo/plugins.xml:8]:
```suggestion
<library path="servo_local_solver">
  <class type="hybrid_planning_demo::ServoSolver" base_class_type="moveit_hybrid_planning::LocalConstraintSolverInterface">
    <description>Uses Servo as a local solver.</description>
  </class>
</library>
<library path="servo_sampler">
  <class type="hybrid_planning_demo::SimpleServoSampler" base_class_type="moveit_hybrid_planning::TrajectoryOperatorInterface">
    <description>Remake of SimpleSampler to use with Servo local solver.</description>
  </class>
</library>
```

**Comment by christianlandgraf** [comment]
It works on the UR10e now! I would suggest to revert the changes in `ipa_demo_cell_description` and I think it's ready to merge aftewards.

**Comment by AdamPettinger** [comment]
> It works on the UR10e now! I would suggest to revert the changes in `ipa_demo_cell_description` and I think it's ready to merge aftewards.

@christianlandgraf Done! Will you verify it looks ok? Note the launch now has to be with the UR10e, or the pre-welding configuration is in collision with the table

@henningkayser I didn't want to update the moveit branch to my fork again since I am leaving soon, but we might want to use your fork rebased on the foxy branch now, and include [this change in hybrid planning](https://github.com/ros-planning/moveit2/pull/488#discussion_r687985278). I hope this will make CI pass

**Review by christianlandgraf — APPROVED** [review]

**Comment by christianlandgraf** [comment]
@AdamPettinger  It looks good to me. I think that's a minor issue since I hope we can mainly continue work with the UR10e from now on...

**Pull request merged**